src/app/login/login.component.ts

```diff
@@ -101,19 +104,57 @@ export class LoginComponent implements OnInit {
                 }
             },
             err => {
-                this.loginerrormessage = 'Error: ' + err.message;
+                if(!this.handleLoginError(err)){
+                    this.loginerrormessage = 'Error: ' + err.message;
+                }
             }
         );
     }
 
+    private handleLoginError(loginresonseobj: any) {
+        const error_msgs_1fa = {
+            429: `
+<p>
+Error: Too many failed logins. You have entered the wrong login details too many times.<br />Please use your unlock code to log in, or contact Runbox Support at support@runbox.com.
+</p>
+`,
+            430: `
+<p>
+Error: Excessive failed logins - You have entered wrong login details excessively.<br />For security reasons you have been temporarily banned from logging in. Try again later, or contact Runbox Support at support@runbox.com.
+</p>
+`,
+            500: `
+<p>
+Error: 500 - Authentication server error. Please contact Runbox Support at support@runbox.com.
+</p>
+`,
+        };
+        if (!loginresonseobj.is_2fa_enabled && loginresonseobj.code && error_msgs_1fa[loginresonseobj.code]) {
+            this.login_error_html = error_msgs_1fa[loginresonseobj.code];
+            return true;
+        }
+        if (loginresonseobj.user_status > 0 && loginresonseobj.user_status < 5 && loginresonseobj.error) {
+            this.login_error_html = loginresonseobj.error;
+            return true;
+        }
+        return false;
+    }
+
     private handleLoginResponse(loginresonseobj: any) {
         if (loginresonseobj.user_status > 0 && loginresonseobj.user_status < 5) {
```

Comment:
this should probably reuse login_errors_reset() – the else is forgetting to reset login_error_html. It's probably safer to reset it initially even if login_error_html is going to be set later.